routes/image_routes.js

```diff
@@ -6,102 +6,205 @@ const db = require('../models');
+  let targetId = req.params.id;
+  let user = req.user;
+  let authorId;
+
+  return Authors.findAll({where: {name: data.author}})
+  .then (authors => {
+    if (authors.length === 0){
+      return Authors.create({name: data.author});
     } else {
-      return result.update({url: data.url, description: data.description});
+      return authors[0];
     }
   })
+  .then(author => {
+    return authorId = author.id;
+  })
+  .then (() => {
+    return Images.findById(targetId, {include: [Authors]});
+  })
+  .then(image => {
+    if (!image){throw new Error('Error - trying to edit an image that does not exist.');};
+    if (image.user_id !== user.id) {throw new Error ('Trying to edit someone else\'s image is not nice.');};
+    return image.update({url: data.url, description: data.description, authors_id: authorId});
+  })
   .then (result => {
-    return res.json(result);
+    return res.redirect('/gallery/'+targetId);
+    // return res.json(result);
   })
-    .catch((error) => {
+  .catch((error) => {
     console.log ('here is our error', error);
-    return res.status(400).send('something went terribly wrong');
+    return res.status(400).send(error.message);
   });
 });
 
-router.get('/:id', (req, res)=>{
-  let id = req.params.id;
-  return Images.findById(id)
-  .then(result =>{
-    if(result === null){
-      res.json('Not today loser');
-    } else{
-    res.json(result);
-    }
-  });
-});
 
-router.delete('/:id', (req, res) =>{
-  let id = req.params.id;
-  return Images.destroy({where: {id: id}})
-  .then(result =>{
-    console.log('results from delete:',result);
-    res.end();
+router.get('/:id/edit', (req, res) => {
+  let targetId = parseInt(req.params.id);
+  let user = req.user;
+
+  let allViewObj = {
+    targetImage: {},
+    otherImages: []
+  };
+
+  return Images.findById(targetId, {include: [ Authors ]})
+  .then(image => {
+    if (!image) {throw new Error('Error - trying to edit an image that does not exist.');};
+    if (image.user_id !== user.id) {throw new Error ('Trying to edit someone else\'s image is not nice.');};
+    return res.render('editForm', image);
+  })
+  .catch((error) => {
+    console.log ('here is our error', error);
+    return res.status(400).send(error.message);
   });
 });
 
 
+router.get('/:id', (req, res) => {
+  let targetId = parseInt(req.params.id);
+  let user = req.user;
 
+  let singleViewObj = {
+    targetImage: {},
+    otherImages: [],
+  };
 
+  return Images.findById(targetId, {include: [ Authors ]})
+  .then(image => {
+    if(!image){
+      throw new Error ('Error - image with that id does not exist');
+    } else {
+      let ownedByUser = (image.user_id === user.id);
+      let targetImage = {id: image.id, url: image.url, description: image.description, user_id: image.user_id, author: image.Author.name, ownedByUser: ownedByUser};
+      return singleViewObj.targetImage = targetImage;
+    }
+  })
+  .then(() => {
+    return Images.findAll({ include: [ Authors ]})
```

Comment:
this is fine. however, there is a better Sequelize way than to do a filter based on `image.id !== targetId`
see if you can do a `Images.find()` `where` **image.id** is not equal to **targetId**, `include` **Authors**.